### PR TITLE
Set timeout to infinity with trace enabled

### PR DESCRIPTION
**Pull request by laurocaetano**
Automatically set the `:timeout` to `:infinity` when trace is enabled.

Also I extracted some logic from the `run` function in the `Runner` module to a new function so it could be tested.

@josevalim thanks again for the help with this patch :)

### CODE REVIEW

File: lib/ex_unit/test/runner_test.exs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Code.require_file "test_helper.exs", __DIR__
+
+defmodule ExUnitTest do
+  use ExUnit.Case
```
Comment (josevalim) [inline lib/ex_unit/test/runner_test.exs]: We can set `async: true` here since this is an unit test. :D

Comment (josevalim) [inline lib/ex_unit/test/runner_test.exs]: Actually, nevermind, we are using the `ExUnit.configuration` and other process could change it.

File: lib/ex_unit/test/runner_test.exs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  test "it set the timeout to infinity and max cases to one with trace enabled"
+    do opts = Keyword.merge(ExUnit.configuration, trace: true)
```
Comment (josevalim) [inline lib/ex_unit/test/runner_test.exs]: Oops, `do` should be in the previous line!

File: lib/ex_unit/test/runner_test.exs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Code.require_file "test_helper.exs", __DIR__
+
+defmodule ExUnitTest do
```
Comment (josevalim) [inline lib/ex_unit/test/runner_test.exs]: Let's change this to `ExUnit.RunnerTest` to mimic the module we are testing? And we should move the file to `test/ex_unit/runner_test.exs`. :)

**Comment by josevalim** [comment]
Awesome, I have added some comments!

File: lib/ex_unit/test/runner_test.exs (hunk 2 above)
Comment (laurocaetano) [inline lib/ex_unit/test/runner_test.exs]: emacs :trollface:

File: lib/ex_unit/test/runner_test.exs (hunk 3 above)
Comment (laurocaetano) [inline lib/ex_unit/test/runner_test.exs]: Bad copy/paste :cry:

**Comment by laurocaetano** [comment]
@josevalim thanks. Updated.

**Pull request merged**